src/components/v5/common/Fields/InputBase/FormattedInput.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { FC, useEffect, useRef, useState } from 'react';
+import Cleave from 'cleave.js/react';
+import { ReactInstanceWithCleave } from 'cleave.js/react/props';
+import clsx from 'clsx';
+
+import { FormattedInputProps } from './types';
+import { useStateClassNames } from '../hooks';
+import { FIELD_STATE } from '../consts';
+
+const displayName = 'v5.common.Fields.FormattedInput';
+
+const FormattedInput: FC<FormattedInputProps> = ({
+  options,
+  onChange,
+  disabled,
+  value,
+  className,
+  state,
+  mode = 'primary',
+  stateClassNames: stateClassNamesProp,
+  buttonProps,
+  wrapperClassName,
+  message,
+  ...rest
+}) => {
+  const [cleave, setCleave] = useState<ReactInstanceWithCleave | null>(null);
+  const wrapperRef = useRef<HTMLDivElement | null>(null);
+  const buttonRef = useRef<HTMLButtonElement | null>(null);
+
+  /**
+   * Sync the cleave raw value with hook-form value
+   * This is necessary for correctly setting the initial value
+   */
```

Comment:
This comment is almost correct. This hooks servers as a sync logic between `value` prop and the internal `Cleave` value. Hook form logic does not exist here yet